src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_SwitchExpression.cs

```diff
@@ -51,11 +57,26 @@ private BoundExpression LowerSwitchExpression(BoundConvertedSwitchExpression nod
                     node.DecisionDag, loweredSwitchGoverningExpression, result, out BoundExpression savedInputExpression);
                 Debug.Assert(savedInputExpression != null);
 
+                object restorePointForEnclosingStatement = new object();
```

Comment:
We only need those identifiers if `produceDetailedSequencePoints` is true. Can we avoid allocations?

Reply:
These two allocations are a drop in the bucket compared to the allocations used to manipulate the decision dag (e.g. line 56).  If we wanted to reduce allocations, this would not be a place to start.

---
In reply to: [398873669](https://github.com/dotnet/roslyn/pull/42711#discussion_r398873669) [](ancestors = 398873669)